metpy/calc/tools.py

```diff
@@ -157,3 +158,35 @@ def interpolate_nans(x, y, kind='linear'):
     else:
         raise ValueError('Unknown option for kind: {0}'.format(str(kind)))
     return y[x_sort_args]
+
+
+@exporter.export
```

Comment:
Fair - Can do. Re above - I don't think we ever need to look back. `nearest_intersection_idx` shouldn't ever return a masked point.